### PR TITLE
Check the existence of the gzipped path explicitly

### PR DESCRIPTION
**Pull request by arunoda**
We couldn't depend on the previous check and it cause errors randomly.
This also fixes issues like https://github.com/zeit/next.js/issues/677

### CODE REVIEW

File: server/render.js [hunk 1]
```diff
@@ -147,17 +148,27 @@ export async function serveStaticWithGzip (req, res, path) {
...
+    // Getting `ENOENT` error from the `serveStatic` is inconsistance and
+    // didn't work on all the cases.
+    //
+    // And this won't give us a race condition because we know that
+    // we don't add gzipped files at runtime.
+    await fs.stat(gzipPath)
   } catch (ex) {
     if (ex.code === 'ENOENT') {
-      res.removeHeader('Content-Encoding')
+      // Seems like there's no gzipped file. Let's serve the uncompressed file.
```
Comment (rauchg) [inline server/render.js]: when would this happen?

Comment (arunoda) [inline server/render.js]: Specially in dev mode.
We can remove this check when we get https://github.com/zeit/next.js/pull/657

**Comment by arunoda** [comment]
@rauchg Check now.

**Comment by rauchg** [comment]
Looking forward to removing the check, doesn't seem necessary

**Pull request merged**